kolibri/core/errorreports/middleware.py

```diff
@@ -1,12 +1,37 @@
 import logging
 import traceback
+from sys import version_info
 
+import pkg_resources
 from django.db import IntegrityError
 
 from .constants import BACKEND
 from .models import ErrorReports
 
 
+def get_request_info(request):
+    return {
+        "url": request.build_absolute_uri(),
+        "method": request.method,
+        "headers": dict(request.headers),
+        "body": request.body.decode("utf-8"),
+    }
+
+
+def get_server_info(request):
+    return {"host": request.get_host(), "port": request.get_port()}
+
+
+def get_packages():
```

Comment:
As we statically bundle most dependencies in Kolibri currently, this may be less informative than its might be in other use cases, but is probably OK for now.

We might also consider caching this, as it shouldn't change over the runtime of a server.